pkg/lib/saml/service.go

```diff
@@ -304,7 +308,65 @@ func (s *Service) IssueSuccessResponse(
 
 	response.Assertion = assertion
 
-	// TODO(saml): Sign the response
+	// Sign the assertion
+	// Reference: https://github.com/crewjam/saml/blob/193e551d9a8420216fae88c2b8f4b46696b7bb63/identity_provider.go#L833
+	signingContext, err := s.idpSigningContext()
+	if err != nil {
+		return nil, err
+	}
+
+	assertionEl := response.Assertion.Element()
+
+	signedAssertionEl, err := signingContext.SignEnveloped(assertionEl)
+	if err != nil {
+		return nil, err
+	}
+
+	assertionSigEl := signedAssertionEl.ChildElements()[len(signedAssertionEl.ChildElements())-1]
+	response.Assertion.Signature = assertionSigEl
+
+	// Sign the response
+	responseEl := response.Element()
+	signedResponseEl, err := signingContext.SignEnveloped(responseEl)
+	if err != nil {
+		return nil, err
+	}
+
+	responseSigEl := signedResponseEl.ChildElements()[len(signedResponseEl.ChildElements())-1]
+	response.Signature = responseSigEl
```

Comment:
This code is worth being hosted in its own method. Consider adding some tests for this method.

```
signature, err := s.SignResponse(response)
if err != nil {
  return nil, err
}
response.Signature = signature.

func (s *Service) SignResponse(response) (Signature, error)
```

Reply:
Replace `SignEnveloped` with `ConstructSignature(response, true)`. Do not need to manipulate the child.